### PR TITLE
Major fixes and extra logging

### LINKED ISSUE #1: Bug prevents outer loop from working properly - results affected
The calculation of `error0`, the error for the outer (temperature) loop, was put in before `Tu` was calculated:
https://github.com/cydcowley/DLS-model/blob/52f13d57d377bba2816826e21121986299763971/LRBv21.py#L371-L373

This meant that `error0` was always 0.

Fixing this has been observed to affect the results by 20% or so on the outer and more on the inner, but the impact will likely be problem dependent.

### LINKED ISSUE #2: Bug in initialisation can lead to poor numerical performance in certain cases
The initialisation for impurity fraction comes from the Lengyel model, but the initialisation for density and power are user-set:
https://github.com/cydcowley/DLS-model/blob/52f13d57d377bba2816826e21121986299763971/LRBv21.py#L225-L241

This can lead to the solver struggling in some conditions, particularly when used for very high flux expansion and low connection length.
We should be able to resolve it by rearranging the Lengyel equation for density and power instead.

### PR DESCRIPTION
**Pull request by mikekryjak**
This fixes https://github.com/cydcowley/DLS-model/issues/2 and https://github.com/cydcowley/DLS-model/issues/1. 
The fixes will affect performance - advise to rerun models after updating. 
There are also additional minor fixes to allow running just a single front position for debugging as well as the logging of heat flux and temperature profiles for each iteration.

- [x] Fix outer loop not working
- [x] Improve initialisation
- [x] Minor fixes and logging improvements
- [x] Test

### CODE REVIEW

**Review by cydcowley — APPROVED** [review]
looks good. Didn't realise this extra inner leg processing was even here!

**Comment by mikekryjak** [comment]
@cydcowley ready to go

File: LRBv21.py [hunk 1]
```diff
@@ -231,15 +231,17 @@ def iterate(cvar, Tu):
...
+            nu0_guess = np.sqrt((qpllu0**2 ) /(2*kappa0*cz0 *Tu**2 *integralinterp(Tu)))
+            cvar = nu0_guess
         
         elif control_variable == "power":
-            # If control variable = Power, keep nu0 and cz0 constant
-            # Initialise control variable 1/qradial. This is needed so that
+            # Initialise from Lengyel
             # too high a cvar results in a positive error!
-            cvar = 1/qradial #qpllu0
+            qpllu0_guess = np.sqrt(2*kappa0*cz0*nu0**2*Tu**2*integralinterp(Tu))
```
Comment (cydcowley) [inline LRBv21.py]: I think this qpllu0_guess should just be qpllu0 from input

**Review by cydcowley — CHANGES_REQUESTED** [review]
small change to guess qpll, otherwise looks good for merging

File: LRBv21.py (hunk 1 above)
Comment (mikekryjak) [reply LRBv21.py]: Thanks, this makes sense. I've put in a commit and a comment to this effect.

**Pull request merged**